plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/source/CameraUrlGenerator.java

```diff
@@ -45,31 +47,53 @@ public CameraUrlGenerator(CameraServerSource source) {
     if (frameRate > 0) {
       commands.put("fps", Integer.toString(frameRate));
     }
-    return generateUrls(commands, baseUrls);
+    commands.put("name", source.getName());
+    return generateUrls(commands, baseUrls, source.getName());
   }
 
   @VisibleForTesting
-  static String[] generateUrls(Map<String, String> commands, String[] baseUrls) { // NOPMD varargs instead of array
+  static String[] generateUrls(Map<String, String> commands, String[] baseUrls,
+                               String cameraName) {
     if (baseUrls == null || baseUrls.length == 0) {
       return new String[0];
     }
     if (commands.isEmpty()) {
       return baseUrls;
     } else {
-      return Arrays.stream(baseUrls)
-          .map(url -> url + toHttpParams(commands))
+      var urls =  Arrays.stream(baseUrls)
+          .map(url -> toHttpParams(url, commands))
           .toArray(String[]::new);
+      for (var url : urls) {
+        System.out.println(url);
+      }
+      return urls;
     }
   }
 
   @VisibleForTesting
-  static String toHttpParams(Map<String, String> commands) {
+  static String toHttpParams(String input, Map<String, String> commands) {
     if (commands.isEmpty()) {
-      return "";
+      return input;
     }
-    return commands.entrySet().stream()
-        .map(e -> e.getKey() + "=" + e.getValue())
+    // Special case to remove name from LabVIEW camera
+    if (commands.containsKey("name") && commands.get("name").contains("IMAQdx")) {
+      input = input.replaceAll("\\?name=cam\\d", "");
+    }
+    var commandStr = commands.entrySet().stream()
+        .map(e -> {
+          try {
+            return URLEncoder.encode(e.getKey(), "utf-8").replaceAll("\\+", "%20") + "="
```

Comment:
I'd pull the URLEncoder stuff into its own method.

```java
static String httpEncode(String raw)
```

It'd also make this nicer since you wouldn't need the try-catch in the lambda here

Reply:
It can't take a string, it has to take the Map Entry, but yeah I'll do that.

Reply:
I was envisioning something like `return httpEncode(e.getKey()) + "=" + httpEncode(e.getValue())`, but that'd work too

Reply:
I can't do that, because otherwise that `=` gets encoded by the `encode` method. Which we cannot have.